pkg/repository/repository.go

```diff
@@ -2,173 +2,90 @@ package repository
 
 import (
 	"context"
-	"log/slog"
 
-	asyncclient "github.com/metal-stack/metal-apiserver/pkg/async/client"
-	"github.com/metal-stack/metal-apiserver/pkg/db/generic"
 	"github.com/metal-stack/metal-apiserver/pkg/db/metal"
 
 	adminv2 "github.com/metal-stack/api/go/metalstack/admin/v2"
 	apiv2 "github.com/metal-stack/api/go/metalstack/api/v2"
-	ipamv1connect "github.com/metal-stack/go-ipam/api/v1/apiv1connect"
 	mdcv1 "github.com/metal-stack/masterdata-api/api/v1"
-	mdm "github.com/metal-stack/masterdata-api/pkg/client"
-	"github.com/redis/go-redis/v9"
 )
 
 type (
-	Validated[M any] struct {
-		message M
-	}
-	Repository[E Entity, M Message, C CreateMessage, U UpdateMessage, Q Query] interface {
+	Repository[S any, E Entity, M Message, C CreateMessage, U UpdateMessage, Q Query] interface {
```

Comment:
S should have a meaningful name, either with a type alias, or at least with documentation

Reply:
I called it `R` and `Repo` now.